---
id: register-exception
title: "Register an exception"
description: "The dApp should call this method when it cannot proceed with the request processing after an exception happens."
sidebar_label: "Register an exception"
hide_title: true
hide_table_of_contents: true
api: eJztWFtv67gR/isD9kEJqsiOtweLGpsC6dkUDbB7Njhx0QKxkTOWxhY3FKklqdiu4f++GFKy5CRnbw/bPqxfLA3J4Tcf58LRXpiaLHpp9G0hpsLSWjpP9mabU81SkYqCXG5lfJuKWUlQXNc1uNI0qoAclQJfSgcV+dIUsClJg/SQo9bGQ21NTlTARvoSfElg6YeGXDvgnNRrwJUnC6iBum2hxLom7bK5ng10t3suKWhS6Hw3Qs9kAxYqYLkLwwHlEvMn0oxKKjrZswOSzfVc/5tB20ZrHpKaJxZNHpBUpqA0KHyP1pOTcE+Wd/sWNa7JQo2NI3cy41vMS6kJUBdgGZqP46S9tATOoycwqyCr2rnewKZEz9Rt0MGSVsYyWznJZ0Y14C6b6w8G/jmb3QVVjYPcFAQbqRRz40h7MLZdS8WbFpaGyTva9tEo1dRRZ2ufJd9YHYEPt5mMx9lc32pYGl9Cjo5c1CF13fgexZOs69ODR2c03Pzn/c3d7Pa7D4GdwMrx1CtyDte9JStjN2iLYIJIRWv+302xE9O9yI32pD0/Yl0rmQc/Hn3v2FH3wuUlVchPfleTmAqz/J5yL1JRW/Z6L8nxaI07ZbAYTHTeSr1+0/XbySAdk8job3xJlpoKStrCUmq0O8ZdoQ/OS7CS1nnwGwN5iRZzT9YBWoJkvE1gZZQym+i1NUrr2C9K2mJBuaxQgW6qJa/w7By5sZZcbZg5A0YTLHeesrn+h7EgtfOoc0pZc0FYLIlWyWCN40V4tCGcjCK99iX8JZwGK3MwmUxSuPzyixQu/zrmp3fZXF9roKr2uyEDlmpL7G190EXqgmnxzGiLVa2Y1R6TSIWXPgjvWvIPh17WJ5/DgeURvYunNRmP+e/0ZK5zXkEv3cmX1mwyceCDXGGj/OuVN9aGSIk7ZCIdepWnrR/VCuXb/vQZN/maPErOQxR0tz6dnXAxGyTCPOQ0zpVLgkYXZJ03pmc0RmPRhBRRoWLnogLcTnvcDqgMpgTKDqmIaVFMRW1ccHn0pZiKEQ0Se9TsxPRhLxqrxFSU3tduOhp9VWBdP9qQEx4bq/4mDulwznQ0UiZHxUlk+m48fjeKc8VhkQqOv499pN50Rg8ibegLh1RIvTKB19aSLo0Ok9L13a1IBeONLI+zd9mlSIWSOWkX1GusePV1jXlJF5NsLNIXdm02mwzDcGbsetSudaNvbt/ffLi/4TVZ6SsVXebEw+5uYwQO0/wbSVOyrRVp70LWvY05Ymm2sCRlNiFRWuLgCeUuUMMRj6flKmx1LCsveAiaP336NNeyqo31nSe5o8DtHM+JhwJX/J6hXT8/XC5YXtAK8pLyp8eY1x85r591QXA+nWsAALnq42IwL3iq1GBRr+lsMh6n8MV43C3iX22l9merJDjkFKR+RiWLkxqyf0vzITnvlTBk2kp/dtkKYzk6QmI7VlJLV8IV7JOoJplCgiEXJIe5jjV/ZhtqwUVgyT3pgpNUXN5tauHqTVIitRnH0VlL6J8hGbWLU2B3v4pv5+c9cyeUXV3BZDx5RVHywUDdgmlVt/ul4O0uXFPWKPWQF05PUjfB/gA7BmmXTK7AZozobADlZMZD0j48chZLFowtweKZC0ewnJLXODvGnk2Tl2SHeFoRH0Iv5F9SkPNSh5qcTF+hKNBjsnhIKvJ4fHbrR0ecAZNF+kJbmzt+QlM3Y9GvPAx4++VH21nZnm37yof7WWK08TKnIS9REnzzV0D/bYDb3Vu88e0n4VriNDGEGyW/D9x29xZufDuFG8PpoYvqBVwd47qbRuqXubbUrqbc/6xrtwTUFO5RvKSLxf9Pmkg5em3NjG88g3RyLPdDI/oL0m+347Oo+x1b4EfBAMHSEj61NSwUyVf9llz1PRyXgWEh5Ku39BKV/G/ILhALdewaug4WsCubx+tkNtf3RPDQY1yc/QlrefF1vB1evOp+z7si/qJbOr0AMJ6VRedtk/smVncukrhU4c5WkGd3Ol4fjkaFvubW8/ym5pnH0YKeSXGTwlJLFwobnZfHcb5aH9Jwt6tQD+4+H4/WD5rplzfUQef0Rzv/Rzv/P2vn2zZq0GUd2gv7vu1XHkTvxItUsJEs3O+X6OhfVh0OLP6hIbsT04dFKp7RSg47fjukoiQsyIYG54l23FlEx7+Y8dY8XTUM4dXnA2534orYWg7mDuEuBo3W3Xf3M5GKZftxgs+CP6jhhj9c4EZMhUiFCbaERjbI9kKhXje45rlxZ/79CEWjzwc=
sidebar_class_name: "post api-method"
info_path: cartesi-rollups/_versioned_docs/version-2.0/api-reference/rollup/cartesi-rollup-http-api
custom_edit_url: null
---

import MethodEndpoint from "@theme/ApiExplorer/MethodEndpoint";
import ParamsDetails from "@theme/ParamsDetails";
import RequestSchema from "@theme/RequestSchema";
import StatusCodes from "@theme/StatusCodes";
import OperationTabs from "@theme/OperationTabs";
import TabItem from "@theme/TabItem";
import Heading from "@theme/Heading";

<Heading
  as={"h1"}
  className={"openapi__heading"}
  children={"Register an exception"}
>
</Heading>

<MethodEndpoint
  method={"post"}
  path={"/exception"}
  context={"endpoint"}
>
  
</MethodEndpoint>



The dApp should call this method when it cannot proceed with the request processing after an exception happens.
This method should be the last method ever called by the dApp backend while processing a request.

When running in production mode, the Cartesi Server Manager pauses the Cartesi Machine and reverts the entire state of the machine to what it was before receiving the request.
No HTTP status code will be sent or received.

When running in host mode, the Rollup HTTP Server returns the status code 200.
In both cases, the input will be skipped with the reason EXCEPTION and the exception message will be forwarded.


<Heading
  id={"request"}
  as={"h2"}
  className={"openapi-tabs__heading"}
  children={"Request"}
>
</Heading>

<ParamsDetails
  parameters={undefined}
>
  
</ParamsDetails>

<RequestSchema
  title={"Body"}
  body={{"content":{"application/json":{"schema":{"type":"object","properties":{"payload":{"type":"string","description":"The payload is in the Ethereum hex binary format.\nThe first two characters are '0x' followed by pairs of hexadecimal numbers that correspond to one byte.\nFor instance, '0xdeadbeef' corresponds to a payload with length 4 and bytes 222, 173, 190, 175.\nAn empty payload is represented by the string '0x'.\n","example":"0xdeadbeef","title":"Payload"}},"title":"Exception"}}}}}
>
  
</RequestSchema>

<StatusCodes
  id={undefined}
  label={undefined}
  responses={{"200":{"description":"Accepted the exception throw."},"default":{"description":"Error response.","content":{"text/plain":{"schema":{"type":"string","description":"Detailed error message.","example":"The request could not be understood by the server due to malformed syntax","title":"Error"}}}}}}
>
  
</StatusCodes>


      